### PR TITLE
add word.py

### PR DESCRIPTION
**Pull request by ariboyarsky**
Adds Word.py to Word repo. This is the class that will encapsulate nltk functionality relating to individual words, such as retriving definitions, synonyms, antonyms, and POS tags.

### CODE REVIEW

File: Word.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        self.antonyms = []
+        self.wnset = ""
+        self.pos = ""
+        self.definitionList = {}
+        self.getPOS(self)
+        self.getSynAnt(self)
+
+    #gets the nltk POS and then uses the mapper function to map to wordnet POS
+    @staticmethod
+    def getPOS(word):
```
Comment (eliandrew) [inline Word.py]: change to get_pos
Apparantly all python methods are supposed to be all lower case

File: Word.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        tag = nltk.pos_tag(word.text.split(" "))[0][1]
+        word.pos = word.wnposmapper(tag)
+
+    #maps nltk POS tags to wordnet POS tags
+    def wnposmapper(self, tag):
```
Comment (eliandrew) [inline Word.py]: change to wn_pos_mapper

File: Word.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        elif tag.startswith('R'):
+            return wordnet.ADV
+        elif tag.startswith('S'):
+            return wordnet.ADJ_SAT
+        else:
+            return ''
+
+    #getSynAnt will use nltk wordnet to find synonyms and antonyms
+    @staticmethod
+    def getSynAnt(word):
```
Comment (eliandrew) [inline Word.py]: change to get_syn_ant

File: Word.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        #get synsets for word
+        ssets = wordnet.synsets(word.text)
+
```
Comment (eliandrew) [inline Word.py]: remove spaces

File: Word.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        for syns in ssets:
+            for synonym in syns.lemmas():
+                word.synonyms.append(synonym.name())
+
```
Comment (eliandrew) [inline Word.py]: remove spaces

File: Word.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                word.definitionList[syns] = syns.definition()
+
```
Comment (eliandrew) [inline Word.py]: remove spaces

File: Word.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        elif tag.startswith('N'):
+            return wordnet.NOUN
+        elif tag.startswith('R'):
+            return wordnet.ADV
+        elif tag.startswith('S'):
+            return wordnet.ADJ_SAT
+        else:
+            return ''
+
+    # getSynAnt will use nltk wordnet to find synonyms and antonyms
```
Comment (eliandrew) [inline Word.py]: this still has the old name

File: Word.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    @staticmethod
+    def get_syn_ant_def(word):
+        # get synsets for word
+        ssets = wordnet.synsets(word.text)
+        for syns in ssets:
+            for synonym in syns.lemmas():
+                word.synonyms.append(synonym.name())
+                word.definitionList[syns] = syns.definition()
+                if synonym.antonyms():
+                    word.antonyms.append(synonym.antonyms()[0].name())
```
Comment (eliandrew) [inline Word.py:49]: probably add new line at the end of the file so we dont get that weird red mark

**Review by eliandrew — CHANGES_REQUESTED** [review]
just two last small comments otherwise looks gooooood

**Comment by ariboyarsky** [comment]
Changes made.

**Review by eliandrew — APPROVED** [review]

**Pull request merged**